javalib/src/main/scala/java/util/concurrent/ThreadLocalRandom.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Written by Doug Lea with assistance from members of JCP JSR-166
+ * Expert Group and released to the public domain, as explained at
+ * http://creativecommons.org/publicdomain/zero/1.0/
+ *
+ * and translated to Scala
+ */
+
+package java.util.concurrent
+
+import java.util.Random
+
+class ThreadLocalRandom private (var initialized: Boolean = false)
+    extends Random {
+
+  override def setSeed(seed: Long): Unit = {
+    if (initialized)
+      throw new UnsupportedOperationException()
+
+    initialized = true
+    super.setSeed(seed)
+  }
+
+  def nextInt(least: Int, bound: Int): Int = {
+    if (least >= bound)
+      throw new IllegalArgumentException()
+
+    val difference = bound - least
+    if (difference > 0)
+      nextInt(difference) + least
+    else
+      nextInt(-difference) - least
```

Comment:
It might not be obvious from my previous comment, but all the arithmetic there is in signed 2's complement. So `[-least, -bound]` is
`{ -least, -least+1, -least+2, ..., Int.MaxValue, Int.MinValue, Int.MinValue+1, ..., -bound-1, -bound }`